packages/engine/Source/Scene/VoxelTraversal.js

```diff
@@ -784,16 +795,28 @@ function postPassesUpdate(
   }
   traverseRecursive(rootNode);
 
+  primitive.statistics.numberOfTilesWithContentReady =
+    loadStateByCount[KeyframeNode.LoadState.LOADED];
+  primitive.statistics.visited = nodeCountTotal;
+
+  const checkEventListeners =
+    primitive.loadProgress.numberOfListeners > 0 ||
+    primitive.allTilesLoaded.numberOfListeners > 0 ||
+    primitive.initialTilesLoaded.numberOfListeners > 0;
+  if (!that._debugPrint && !checkEventListeners) {
+    return;
```

Comment:
If there are no event listeners, can we exit before calling `traverseRecursive`? recursive traversal is not cheap.

Reply:
The way I have it set up now, `statistics.numberOfTilesWithContentReady` and `statistics.visited` rely on the stats gathered in the traversal. Perhaps these stats could be gathered another way during the loadAndUnload function?

Reply:
And unlike events where we can check is event listeners habe subscribed we do not really know if anybody is seeking these stats, unless perhaps we had a flag to turn off/on collection of this data.